<?xml version="1.0" encoding="UTF-8"?>

<launch>

    <arg name="uav_id" default="px4_quad"/>
    <arg name="ugv_id" default="nexus5"/>
    <arg name="run_simulation" default="false"/>

    <node name="UAV_controller" pkg="Rendezvous" type="UAVoffboard_node"/>
    <node name="UGV_controller" pkg="Rendezvous" type="distributed_UGVoffboard_node">
        <remap from="/qualisys/nexus1/pose" to="/qualisys/$(arg ugv_id)/pose"/>
        <remap from="/qualisys/nexus1/velocity" to="/qualisys/$(arg ugv_id)/velocity"/>
        <remap from="/nexus1/cmd_vel" to="/$(arg ugv_id)/cmd_vel"/>
	<remap from="/nexus1/trajectory" to="/$(arg ugv_id)/trajectory"/>
    </node>

    <group unless="$(arg run_simulation)">
        <!-- TODO -->
        <node name="record" pkg="rosbag" type="record" args="-j -O rendezvous.bag -a"/>
    </group>

    <!-- Run simulation if argument is set-->
    <group ns="simulation" if="$(arg run_simulation)">
        <node name="simulator" pkg="Rendezvous" type="simulator_node"> <!--output="screen"-->
            <remap from="/qualisys/nexus1/pose" to="/qualisys/$(arg ugv_id)/pose"/>
            <remap from="/qualisys/nexus1/velocity" to="/qualisys/$(arg ugv_id)/velocity"/>
            <remap from="/nexus1/cmd_vel" to="/$(arg ugv_id)/cmd_vel"/>
	    <remap from="/nexus1/initial_pose" to="/$(arg ugv_id)/initial_pose"/>
        </node>
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find Rendezvous)/rendezvous.rviz"/>
        <node name="record" pkg="rosbag" type="record" args="-j -O rendezvous_simulation.bag -a"/>
    </group>

</launch>
